src/storage.rs

```diff
@@ -33,11 +39,30 @@ pub enum OwnedStorage<const N: usize> {}
 impl<const N: usize> Storage for OwnedStorage<N> {}
 impl<const N: usize> SealedStorage for OwnedStorage<N> {
     type Buffer<T> = [T; N];
+    fn len<T>(_: *const Self::Buffer<T>) -> usize {
+        N
+    }
+    fn as_ptr<T>(this: *mut Self::Buffer<T>) -> *mut T {
+        this as _
+    }
 }
 
 /// Implementation of [`Storage`] that stores the data in an unsized `[T]`.
 pub enum ViewStorage {}
 impl Storage for ViewStorage {}
 impl SealedStorage for ViewStorage {
     type Buffer<T> = [T];
+    fn len<T>(this: *const Self::Buffer<T>) -> usize {
+        // We get the len of the buffer. There is no reverse method of `core::ptr::from_raw_parts`, so we have to work around it.
```

Comment:
Rust 1.79 has stabilized `[T]::len()`, so you can do `this.len()`.

Reply:
Oh, that's good!

Reply:
Done in 22efa13bf24aafe5bba28751443fca7454e2069d